### PR TITLE
Support reloading consuming segment using force commit

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Another try of #9310 

- Use force commit to achieve consuming segment reload
- Modify HybridClusterIntegrationTest to use LLC to test the consuming segment reload
- Make LLC default for integration test because most (if not all) use cases use LLC
- Also fix a bug for adding empty segment to dedup table

### CODE REVIEW

**Review by sajjad-moradi — APPROVED** [review]
LGTM

**Comment by jugomezv** [comment]
> Another try of #9310
> 
> * Use force commit to achieve consuming segment reload
> * Modify HybridClusterIntegrationTest to use LLC to test the consuming segment reload
> * Make LLC default for integration test because most (if not all) use cases use LLC

@Jackie-Jiang I will go ahead and delete the PR I started as you pushed the change here. Thanks for driving it as I have been real busy at work these days

**Comment by Jackie-Jiang** [comment]
@jugomezv Thanks for trying it out. I didn't realize it requires changing the integration test in the beginning

**Review by siddharthteotia — APPROVED** [review]

File: pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java [hunk 1]
```diff
@@ -373,16 +369,28 @@ private void reloadSegmentWithMetadata(String tableNameWithType, SegmentMetadata
...
           return;
         }
+        // TODO: Support force committing HLC consuming segment
+        if (!(segmentDataManager instanceof LLRealtimeSegmentDataManager)) {
+          LOGGER.warn("Cannot reload non-LLC consuming segment: {} in table: {}", segmentName, tableNameWithType);
+          return;
+        }
+        LOGGER.info("Reloading (force committing) LLC consuming segment: {} in table: {}", segmentName,
+            tableNameWithType);
+        ((LLRealtimeSegmentDataManager) segmentDataManager).forceCommit();
```
Comment (siddharthteotia) [inline pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java:392]: So forceCommit is async ?

Comment (Jackie-Jiang) [reply pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java:392]: Yes, it is async. We can only signal the segment to perform the commit

**Pull request merged**